civiform/civiform
# Warn users if they upload a file that's too large

### Description

If a user uploads a file > 100MB to a file upload question, display an error saying that the file is too large and prevent the form from submitting.

This helps prevent users from seeing an AWS error and being taken away from the CiviForm site (see screenshots in attached bug). Note that the size check has to be done client-side since the file is immediately uploaded to AWS and is never uploaded to CiviForm's server.

## Release notes

### Checklist

#### General

Read the full guidelines for PRs [here](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#creating-a-pull-request)


#### User visible changes


### Instructions for manual testing

1. Upload a file larger than 100MB to a file upload question.
2. Verify the error immediately appears.
3. Click "Save & next".
4. Verify the error remains and the page doesn't save the file.
5. Upload a new file smaller than 100MB.
6. Verify the error disappears.
7. Click "Save & next".
8. Verify the file is saved and you proceed to the next block.

### Issue(s) this completes

Fixes #5675
